pkg/relay/submit.go

```diff
@@ -75,6 +75,12 @@ func (rs *Relay) SubmitBlock(ctx context.Context, m *structs.MetricGroup, submit
 }
 
 func (rs *Relay) isPayloadDelivered(ctx context.Context, slot uint64) (err error) {
+	// check if delivered in stream layer
+	if rs.isPayloadDeliveredInStream(slot) {
```

Comment:
as above - I dont think we need anything more than just `slot <= uint64(rs.slotDelivered)` here